### PR TITLE
QOL Improvment: Remove dependency on attribute and use property instead for keychain password in certificate resource

### PR DESCRIPTION
**Pull request by jazaval**
## Describe the Pull Request
When using the certificate resource install a certificate to a locked keychain, the password to unlock is only settable via the node['macos']['admin_password'] attribute.

This means users of the certificate resource will have to re-declare this above any calls to the resource unless it's in kitchen.

This PR, remove that logic and instead introduces a kc_passwd property similar to the keychain resource to allow users to input this password.

## Justification
Not really a bug but a QOL improvement.

### CODE REVIEW

**Comment by mjmerin** [comment]
LGTM

**Review by mjmerin — APPROVED** [review]

**Review by v-anshie — APPROVED** [review]

File: resources/certificate.rb [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  cert = SecurityCommand.new(new_resource.certfile, keychain)
+  cert = SecurityCommand.new(new_resource.certfile, new_resource.keychain)
 
   execute 'unlock keychain' do
-    command Array(cert.unlock_keychain(node['macos']['admin_password']))
+    command Array(cert.unlock_keychain(new_resource.kc_passwd))
   end
 
+  cert_shasum = shell_out("shasum #{new_resource.certfile}").stdout.upcase.gsub(/\s.+/, '')
+  find_cert_output = shell_out("/usr/bin/security find-certificate -a -Z #{new_resource.keychain}").stdout
```
Comment (jweyer8) [inline resources/certificate.rb:20]: `/usr/bin/security find-certificate -a -Z | awk '/SHA/ {print $3}'` might optimize the `include?` a bit?

**Review by jweyer8 — APPROVED** [review]

File: resources/certificate.rb [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 action :install do
-  cert = SecurityCommand.new(new_resource.certfile, keychain)
+  cert = SecurityCommand.new(new_resource.certfile, new_resource.keychain)
 
   execute 'unlock keychain' do
-    command Array(cert.unlock_keychain(node['macos']['admin_password']))
+    command Array(cert.unlock_keychain(new_resource.kc_passwd))
   end
 
+  cert_shasum = shell_out("shasum #{new_resource.certfile}").stdout.upcase.gsub(/\s.+/, '')
```
[the remark this answers is not in the document]
Comment (mjmerin) [reply resources/certificate.rb:19]: .strip would only remove the white space, what I'm doing here is also taking out anything after the whitespace.

File: resources/certificate.rb (hunk 1 above)
Comment (mjmerin) [reply resources/certificate.rb:20]: I believe that finds the shasums of all certificates on all keychains. So depending on how many certs/keychains there are, this more direct approach is probably faster and easier to read. Again, I don't really have any metrics to confirm or deny.

**Pull request merged**